chaoscenter/authentication/api/middleware/jwt_middleware_test.go

```diff
@@ -0,0 +1,234 @@
+	panic("unimplemented")
+}
+
+// RemoveInvitation implements services.ApplicationService.
+func (m *MockApplicationService) RemoveInvitation(projectID string, userID string, invitation entities.Invitation) error {
+	panic("unimplemented")
+}
+
+// RevokeToken implements services.ApplicationService.
+func (m *MockApplicationService) RevokeToken(tokenString string) error {
+	panic("unimplemented")
+}
+
+// UpdateInvite implements services.ApplicationService.
+func (m *MockApplicationService) UpdateInvite(projectID string, userID string, invitation entities.Invitation, role *entities.MemberRole) error {
+	panic("unimplemented")
+}
+
+// UpdatePassword implements services.ApplicationService.
+func (m *MockApplicationService) UpdatePassword(userPassword *entities.UserPassword, isAdminBeingReset bool) error {
+	panic("unimplemented")
+}
+
+// UpdateProjectName implements services.ApplicationService.
+func (m *MockApplicationService) UpdateProjectName(projectID string, projectName string) error {
+	panic("unimplemented")
+}
+
+// UpdateProjectState implements services.ApplicationService.
+func (m *MockApplicationService) UpdateProjectState(ctx context.Context, userID string, deactivateTime int64, isDeactivate bool) error {
+	panic("unimplemented")
+}
+
+// UpdateStateTransaction implements services.ApplicationService.
+func (m *MockApplicationService) UpdateStateTransaction(userRequest entities.UpdateUserState) error {
+	panic("unimplemented")
+}
+
+// UpdateUser implements services.ApplicationService.
+func (m *MockApplicationService) UpdateUser(user *entities.UserDetails) error {
+	panic("unimplemented")
+}
+
+// UpdateUserState implements services.ApplicationService.
+func (m *MockApplicationService) UpdateUserState(ctx context.Context, username string, isDeactivate bool, deactivateTime int64) error {
+	panic("unimplemented")
+}
+
+func (m *MockApplicationService) ValidateToken(tokenString string) (*jwt.Token, error) {
+	return nil, nil
+}
+
+func TestJwtMiddleware(t *testing.T) {
+	// Initialize Gin Engine
+	router := gin.Default()
+
+	// Set up the middleware with the mock service
+	mockService := &MockApplicationService{}
+	router.Use(middleware.JwtMiddleware(mockService))
+
+	// Define a test route that uses the middleware
+	router.GET("/status", func(c *gin.Context) {
+		c.JSON(http.StatusOK, gin.H{"message": "Access Granted"})
+	})
+
+	// Helper function to create requests
+	createRequest := func(token string) *httptest.ResponseRecorder {
+		req := httptest.NewRequest("GET", "/status", nil)
+		req.Header.Set("Authorization", "Bearer "+token)
+		w := httptest.NewRecorder()
+		return w
+	}
+
+	// Subtest for Valid Token
+	t.Run("Valid Token", func(t *testing.T) {
+		wValid := createRequest("jwtstring")
+		assert.Equal(t, http.StatusOK, wValid.Code)
+	})
+
+	// Subtest for Missing Authorization Header
+	t.Run("Missing Authorization Header", func(t *testing.T) {
```

Comment:
I see there's 30% code coverage with this TC, please check if you can add more TCs for more code coverage